### PR TITLE
Add to_json() for std::vector<bool>::reference

### PR DESCRIPTION
**Pull request by falbrechtskirchinger**
Avoids surprises as seen in #3533.

### CODE REVIEW

**Comment by falbrechtskirchinger** [comment]
Changed to `static_cast<>()`. Should be good now.

**Review by nlohmann — APPROVED** [review]
Looks good to me.

File: tests/src/unit-constructor1.cpp [hunk 1]
```diff
@@ -474,6 +474,13 @@ TEST_CASE("constructors")
             json j(false);
             CHECK(j.type() == json::value_t::boolean);
         }
+
+        SECTION("from std::vector<bool>::refrence")
```
Comment (mdrost) [inline tests/src/unit-constructor1.cpp:478]: typo in "refrence" :P

Comment (falbrechtskirchinger) [reply tests/src/unit-constructor1.cpp:478]: Thanks, I'll add it to #3643.

**Pull request merged**